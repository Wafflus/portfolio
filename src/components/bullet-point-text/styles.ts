import styled from "styled-components";

export const BulletPointTextStyled = styled.li`
    color: ${({ theme }) => theme.data.text.lightColor};
    font-family: '${({ theme }) => theme.data.fontFamilies.outfit}', sans-serif;
    font-size: 1.125rem;
    letter-spacing: 0.1rem;
    line-height: 120%;
    list-style-type: none;
    position: relative;
    text-align: center;
    transition: transform 0.5s;

    &:hover {
        transform: scale(102%);
    }
    
    ${props => props.theme.breakpoint.above('lg',
        `
            &::before {
                background-color: ${props.theme.data.secondaryColor};
                border-radius: 50%;
                content: '';
                height: 20px;
                left: -40px;
                position: absolute;
                transition: background-color 0.5s;
                width: 20px;
            }

            &:hover::before {
                background-color: ${props.theme.data.primaryColor};
            }
        `
    )}
`;