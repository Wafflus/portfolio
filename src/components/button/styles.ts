import { IButtonStyledProps } from "./types";
import styled, { css } from "styled-components";

export const ButtonStyled = styled.a<IButtonStyledProps>`
    align-items: center;
    background-color: ${({ theme }) => theme.data.secondaryColor};
    border-radius: 8px;
    color: ${({ theme }) => theme.data.text.darkColor};
    display: flex;
    font-size: 2rem;
    font-variant: small-caps;
    gap: 16px;
    height: 66px;
    justify-content: center;
    letter-spacing: 0.1rem;
    max-width: 300px;
    min-width: 200px;
    padding: 8px;
    text-decoration: none;
    transition: transform 0.5s;
    width: 100%;

    ${props => props.outline && css`
        background-color: transparent;
        border: 1px solid ${({ theme }) => theme.data.primaryColor};
        color: #93855B;
    `}

    &:hover {
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
        transform: scale(102%);
    }
`;