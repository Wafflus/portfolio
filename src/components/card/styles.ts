import styled from "styled-components";

export const CardStyled = styled.article`
    border-radius: 8px;
    padding: 16px;

    &:nth-child(odd) {
        background-color: ${({ theme }) => theme.data.highlightColor};
    }

    &:hover {
        outline: 3px solid ${({ theme }) => theme.data.borderColor};

        & > div > img {
            border: 3px solid ${({ theme }) => theme.data.primaryColor};
            transform: scale(102%);
        }
    }

    ${props => props.theme.breakpoint.above('lg',
        `
            &:nth-child(even) > div {
                flex-direction: row-reverse;
            }
        `
    )}
`;

export const TitleStyled = styled.h3`
    color: ${({ theme }) => theme.data.text.darkColor};
    font-size: 1.5rem;
    font-variant: small-caps;
    font-weight: 600;
    letter-spacing: 0.1rem;
    text-align: center;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            font-size: 2rem;
            text-align: start;
        `
    )}
`;

export const ToolsStyled = styled.h4`
    color: ${({ theme }) => theme.data.text.defaultColor};
    font-variant: small-caps;
    letter-spacing: 0.1rem;
    text-align: center;
`;

export const DescriptionStyled = styled.p`
    color: ${({ theme }) => theme.data.text.defaultColor};
    flex: 1;
    letter-spacing: 0.1rem;
`;