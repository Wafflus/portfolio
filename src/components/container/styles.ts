import { IContainerStyledProps } from "./types";
import styled from "styled-components";

export const ContainerStyled = styled.div<IContainerStyledProps>`
    height: 100%;
    max-width: 1440px;
    padding: ${props => props.verticalPadding} ${props => props.horizontalPadding};
    width: 100%;
    z-index: 200;
`;