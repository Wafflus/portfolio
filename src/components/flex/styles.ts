import { IFlexStyledProps } from "./types";
import styled from "styled-components";

export const FlexStyled = styled.div<IFlexStyledProps>`
    align-items: ${props => props.defaultFlexProps?.alignItems};
    display: flex;
    flex-direction: ${props => props.defaultFlexProps?.flexDirection};
    gap: ${props => props.defaultFlexProps?.gap};
    justify-content: ${props => props.defaultFlexProps?.justifyContent};
    max-width: ${props => props.defaultFlexProps?.maxWidth};
    width: 100%;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            align-items: ${props.largeBreakpointFlexProps?.alignItems};
            flex-direction: ${props.largeBreakpointFlexProps?.flexDirection};
            gap: ${props.largeBreakpointFlexProps?.gap};
            justify-content: ${props.largeBreakpointFlexProps?.justifyContent};
            max-width: ${props.largeBreakpointFlexProps?.maxWidth};
        `
    )}
`;