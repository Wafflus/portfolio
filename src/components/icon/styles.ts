import styled from "styled-components";

import { IImageProps } from "./types";

export const IconStyled = styled.img<IImageProps>`
    width: ${props => props.size}${props => props.unit};
    height: ${props => props.size}${props => props.unit};
    border-radius: 50%;
`;