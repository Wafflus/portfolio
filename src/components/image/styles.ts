import styled from "styled-components";

export const ImageStyled = styled.img`
    border-radius: 8px;
    border: 3px solid transparent;
    height: 100%;
    max-height: 350px;
    max-width: 500px;
    transition: transform 0.5s, border 0.5s;
    width: 100%;
`;