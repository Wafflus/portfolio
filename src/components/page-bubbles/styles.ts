import styled from "styled-components";
import { IBubbleStyledProps } from "./types";

export const PageBubbleStyled = styled.div<IBubbleStyledProps>`
    opacity: 0;
    
    &:nth-child(1),
    &:nth-child(2) {
        bottom: -100%;
        left: -100%;
    }

    &:nth-child(3),
    &:nth-child(4),
    &:nth-child(5) {
        bottom: -100%;
    }

    ${props => props.theme.breakpoint.above('xl',
        `
            background-color: ${props.color};
            border-radius: 50%;
            opacity: 1;
            position: absolute;
            transition: opacity 0.5s, bottom 0.5s, left 0.5s;
        
            &:nth-child(1),
            &:nth-child(2) {
                height: 293px;
                width: 293px;
            }
        
            &:nth-child(3),
            &:nth-child(5) {
                height: 330px;
                width: 330px;
            }
        
            &:nth-child(4) {
                height: 282px;
                width: 282px;
            }
        
            &:nth-child(1) {
                bottom: -30%;
                left: -115px;
                z-index: 10;
            }
        
            &:nth-child(2) {
                bottom: -45%;
                left: 90px;
                z-index: 20;
            }
        
            &:nth-child(3) {
                bottom: -70%;
                left: calc(100% - 330px - 90px);
                z-index: 10;
            }
        
            &:nth-child(4) {
                bottom: -50%;
                left: calc(100% - 282px + 50px);
                z-index: 20;
            }
        
            &:nth-child(5) {
                bottom: -25%;
                left: calc(100% - 330px + 210px);
                z-index: 10;
            }
        `
    )}
`;