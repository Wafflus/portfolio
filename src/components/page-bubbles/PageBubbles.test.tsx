import "jest-styled-components";
import { breakpoints, theme } from "styles";
import { PageBubbles } from ".";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

test('page bubbles renders', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <PageBubbles pageBubbleStyleProps={{ primaryColor: theme.data.primaryColor, secondaryColor: theme.data.secondaryColor }} />
        </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toBeInTheDocument();
});

test('should have 5 bubbles', () => {
    const pageBubbles = render(
        <ThemeProvider theme={theme}>
            <PageBubbles pageBubbleStyleProps={{ primaryColor: theme.data.primaryColor, secondaryColor: theme.data.secondaryColor }} />
        </ThemeProvider>
    );

    const bubbles = pageBubbles.getAllByRole('presentation');

    expect(bubbles).toHaveLength(5);
})

test('should have correct color', () => {
    const pageBubbles = render(
        <ThemeProvider theme={theme}>
            <PageBubbles pageBubbleStyleProps={{ primaryColor: theme.data.primaryColor, secondaryColor: theme.data.secondaryColor }} />
        </ThemeProvider>
    );

    const bubbles = pageBubbles.getAllByRole('presentation');

    const options = { media: `(min-width: ${breakpoints.xl}px)` };

    expect(bubbles[0]).toHaveStyleRule('background-color', theme.data.primaryColor, options);
    expect(bubbles[1]).toHaveStyleRule('background-color', theme.data.secondaryColor, options);
    expect(bubbles[2]).toHaveStyleRule('background-color', theme.data.secondaryColor, options);
    expect(bubbles[3]).toHaveStyleRule('background-color', theme.data.primaryColor, options);
    expect(bubbles[4]).toHaveStyleRule('background-color', theme.data.secondaryColor, options);
});

test('should only show on extra large (xl) screens', () => {
    const pageBubbles = render(
        <ThemeProvider theme={theme}>
            <PageBubbles pageBubbleStyleProps={{ primaryColor: theme.data.primaryColor, secondaryColor: theme.data.secondaryColor }} />
        </ThemeProvider>
    );

    const bubbles = pageBubbles.getAllByRole('presentation');

    const options = { media: `(min-width: ${breakpoints.xl}px)` };

    expect(bubbles[0]).toHaveStyleRule('opacity', '0');
    expect(bubbles[0]).toHaveStyleRule('opacity', '0');
    expect(bubbles[2]).toHaveStyleRule('opacity', '0');
    expect(bubbles[3]).toHaveStyleRule('opacity', '0');
    expect(bubbles[4]).toHaveStyleRule('opacity', '0');

    expect(bubbles[0]).toHaveStyleRule('opacity', '1', options);
    expect(bubbles[1]).toHaveStyleRule('opacity', '1', options);
    expect(bubbles[2]).toHaveStyleRule('opacity', '1', options);
    expect(bubbles[3]).toHaveStyleRule('opacity', '1', options);
    expect(bubbles[4]).toHaveStyleRule('opacity', '1', options);
});