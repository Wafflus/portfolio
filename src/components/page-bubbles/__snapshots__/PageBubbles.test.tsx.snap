// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page bubbles renders 1`] = `
.c0 {
  opacity: 0;
}

.c0:nth-child(1),.c0:nth-child(2) {
  bottom: -100%;
  left: -100%;
}

.c0:nth-child(3),.c0:nth-child(4),
.c0:nth-child(5) {
  bottom: -100%;
}

@media (min-width:1200px) {
  .c0 {
    background-color: #F9D564;
    border-radius: 50%;
    opacity: 1;
    position: absolute;
    -webkit-transition: opacity 0.5s,bottom 0.5s,left 0.5s;
    transition: opacity 0.5s,bottom 0.5s,left 0.5s;
  }

  .c0:nth-child(1),.c0:nth-child(2) {
    height: 293px;
    width: 293px;
  }

  .c0:nth-child(3),.c0:nth-child(5) {
    height: 330px;
    width: 330px;
  }

  .c0:nth-child(4) {
    height: 282px;
    width: 282px;
  }

  .c0:nth-child(1) {
    bottom: -30%;
    left: -115px;
    z-index: 10;
  }

  .c0:nth-child(2) {
    bottom: -45%;
    left: 90px;
    z-index: 20;
  }

  .c0:nth-child(3) {
    bottom: -70%;
    left: calc(100% - 330px - 90px);
    z-index: 10;
  }

  .c0:nth-child(4) {
    bottom: -50%;
    left: calc(100% - 282px + 50px);
    z-index: 20;
  }

  .c0:nth-child(5) {
    bottom: -25%;
    left: calc(100% - 330px + 210px);
    z-index: 10;
  }
}

@media (min-width:1200px) {

}

<div
  class="c0"
  color="#F9D564"
  role="presentation"
/>
`;
